flang/test/Lower/OpenACC/acc-routine.f90

```diff
@@ -2,12 +2,14 @@
 
 ! RUN: bbc -fopenacc -emit-hlfir %s -o - | FileCheck %s
 
-
+! CHECK: acc.routine @acc_routine_16 func(@_QPacc_routine18) bind("_QPacc_routine17" [#acc.device_type<host>], "_QPacc_routine16" [#acc.device_type<multicore>])
+! CHECK: acc.routine @acc_routine_15 func(@_QPacc_routine17) worker ([#acc.device_type<host>]) vector ([#acc.device_type<multicore>])
+! CHECK: acc.routine @acc_routine_14 func(@_QPacc_routine16) gang([#acc.device_type<nvidia>]) seq ([#acc.device_type<host>])
 ! CHECK: acc.routine @acc_routine_10 func(@_QPacc_routine11) seq
 ! CHECK: acc.routine @acc_routine_9 func(@_QPacc_routine10) seq
 ! CHECK: acc.routine @acc_routine_8 func(@_QPacc_routine9) bind("_QPacc_routine9a")
 ! CHECK: acc.routine @acc_routine_7 func(@_QPacc_routine8) bind("routine8_")
-! CHECK: acc.routine @acc_routine_6 func(@_QPacc_routine7) gang(dim = 1 : i32)
+! CHECK: acc.routine @acc_routine_6 func(@_QPacc_routine7) gang(1 : i64)
```

Comment:
Not having the dim keyword makes this ambiguous relative to the acc loop representation of gang.

Reply:
I wanted to clean it up since only dim is allowed on routine.

Reply:
I had a feeling that was the reason. However, it makes the textual representation inconsistent relative to acc loop. I would need to remind myself of context whether it means 1 gang or gang in dimension 1 when I read the IR.

Reply:
My only concern is that we can have multiple values here and we would just repeat the dim keyword over if we want to be consistent with the loop gang. 

```
acc.routine @acc_routine_6 func(@_QPacc_routine7) gang(dim: 1 : i64 [#acc.device_type<nvidia>], dim: 2: i64 [#acc.device_type<host>])
```

Reply:
The keyword repetition is no problem. IMO it just makes it clearer what the value means at each point.

Reply:
Ok I'll make the change